types/shelljs/index.d.ts

```diff
@@ -25,537 +30,1173 @@ export function cd(dir?: string): ShellString;
  */
 export function pwd(): ShellString;
 
-/**
- * Returns array of files in the given path, or in current directory if no path provided.
- * @param  ...paths Paths to search.
- * @return          An array of files in the given path(s).
- */
-export function ls(...paths: Array<string | string[]>): ShellArray;
+export interface ListFunction {
+	/**
+	 * Returns array of files in the given path, or in current directory if no path provided.
+	 *
+	 * @param options Available options:
+	 *        - `-R`: recursive
+	 *        - `-A`: all files (include files beginning with ., except for . and ..)
+	 *        - `-L`: follow symlinks
+	 *        - `-d`: list directories themselves, not their contents
+	 *        - `-l`: list objects representing each file, each with fields containing
+	 *                `ls -l` output fields. See fs.Stats for more info
+	 * @param paths   Paths to search.
+	 * @return        An array of files in the given path(s).
+	 */
+	(options: string, paths: string[]): ShellArray;
+	(options: string, ...paths: string[]): ShellArray;
```

Comment:
The JSDoc comment seems to only be applied to the first signature (right below the comment) and not to this second signature 😞 (Same in other interfaces.)
Is there a way to make it apply to both (without duplicating the comment once more)? (Probably not 😞)

Reply:
Technically, the JSDoc commwnt that is applied in this case is the one from `export const list: ListFunction;`

Reply:
I guess it depends on the editor/IDE. I was trying this in [CodeSandbox][1] and it seemed to show docs for the first signature, but not the second one.

[1]: https://codesandbox.io/